arithmetization/src/main/java/net/consensys/linea/bl/TransactionProfitabilityCalculator.java

```diff
@@ -0,0 +1,143 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.bl;
+
+import java.math.BigDecimal;
+
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.config.LineaTransactionSelectorConfiguration;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.datatypes.Wei;
+import org.slf4j.spi.LoggingEventBuilder;
+
+@Slf4j
+public class TransactionProfitabilityCalculator {
+
+  private final LineaTransactionSelectorConfiguration conf;
+  private final double preComputedValue;
+
+  public TransactionProfitabilityCalculator(final LineaTransactionSelectorConfiguration conf) {
+    this.conf = conf;
+    this.preComputedValue =
+        conf.estimateGasMinMargin() * conf.gasPriceRatio() * conf.verificationGasCost();
+  }
+
+  public Wei profitablePriorityFeePerGas(
+      final Transaction transaction, final Wei minGasPrice, final long gas) {
+    final double txSize = Math.max(0, transaction.getSize() + conf.adjustTxSize());
+    final double compressedSerializedSize = txSize / conf.txCompressionRatio();
+
+    final var profitAt =
+        preComputedValue
+            * txSize
+            * minGasPrice.getAsBigInteger().doubleValue()
+            / (gas * conf.verificationCapacity());
+
+    final var profitAtWei = Wei.ofNumber(BigDecimal.valueOf(profitAt).toBigInteger());
+
+    log.atDebug()
+        .setMessage(
+            "Estimated profitable priorityFeePerGas: {}; estimateGasMinMargin={}, gasPriceRatio={}, "
+                + "verificationGasCost={}, txSize={}, compressedTxSize={}, minGasPrice={}, gas={}, verificationCapacity={}")
+        .addArgument(profitAtWei::toHumanReadableString)
+        .addArgument(conf.estimateGasMinMargin())
+        .addArgument(conf.gasPriceRatio())
+        .addArgument(conf.verificationGasCost())
+        .addArgument(txSize)
+        .addArgument(compressedSerializedSize)
+        .addArgument(minGasPrice::toHumanReadableString)
+        .addArgument(gas)
+        .addArgument(conf.verificationCapacity())
+        .log();
+
+    return Wei.ofNumber(BigDecimal.valueOf(profitAt).toBigInteger());
+  }
+
+  public boolean isProfitable(
+      final String step,
+      final Transaction transaction,
+      final double minGasPrice,
+      final double effectiveGasPrice,
+      final long gas) {
+    final double revenue = effectiveGasPrice * gas;
+
+    final double l1GasPrice = minGasPrice * conf.gasPriceRatio();
```

Comment:
Note: my interpretation of `l1GasPrice = minGasPrice * conf.gasPriceRatio();` is that we consider the cost on L1 our minL2Gas price * ratio. I this this will yield Optimistic lowest cost on L1, which may lead false positive profit?

Given that L2 Gasprice (`effectiveGasPrice` here I think) flutuates regarding the L1Gasprice (updated by the coordinator) shouldn't we consider `effectiveGasPrice` for l1GasPrice? 

Or is this formula to amortize fluctuations?

** @fab-10  I am aware that this fucntion was already like this, just migrated to this new class. I am just raising this thought **

cc: @julien-marchand

Reply:
`effectiveGasPrice` is the actual gas price of the transaction (computed from base fee and priority fee), which doesn't inform much about the actual L1 gas price.

From what I understand minGasPrice is the actual value that fluctuates based on the coordinator info